File: src/main/java/com/lessmarkup/dataobjects/Menu.java
package com.lessmarkup.dataobjects;

import com.lessmarkup.interfaces.data.AbstractDataObject;
import com.lessmarkup.interfaces.recordmodel.RequiredField;

import java.time.OffsetDateTime;
import java.util.OptionalLong;

public class Menu extends AbstractDataObject {
    private String text;
    private String description;
    private String argument;
    private String uniqueId;
    private int order;
    private boolean visible;
    private OffsetDateTime created;
    private OffsetDateTime updated;
    private OptionalLong imageId;

    @RequiredField
    public String getText() {
        return text;
    }
    public void setText(String text) {
        this.text = text;
    }

    @RequiredField
    public String getDescription() {
        return description;
    }
    public void setDescription(String description) {
        this.description = description;
    }

    public String getArgument() {
        return argument;
    }
    public void setArgument(String argument) {
        this.argument = argument;
    }

    public String getUniqueId() {
        return uniqueId;
    }
    public void setUniqueId(String uniqueId) {
        this.uniqueId = uniqueId;
    }

    public int getOrder() {
        return order;
    }
    public void setOrder(int order) {
        this.order = order;
    }

    public boolean isVisible() {
        return visible;
    }
    public void setVisible(boolean visible) {
        this.visible = visible;
    }

    @RequiredField
    public OffsetDateTime getCreated() {
        return created;
    }
    public void setCreated(OffsetDateTime created) {
        this.created = created;
    }

    public OffsetDateTime getUpdated() {
        return updated;
    }
    public void setUpdated(OffsetDateTime updated) {
        this.updated = updated;
    }

    public OptionalLong getImageId() {
        return imageId;
    }
    public void setImageId(OptionalLong imageId) {
        this.imageId = imageId;
    }
}

File: src/main/scala/com/lessmarkup/dataobjects/Menu.scala
/*
 * This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0.
 * If a copy of the MPL was not distributed with this file, You can obtain one at
 * http://mozilla.org/MPL/2.0/.
 */

package com.lessmarkup.dataobjects

import java.time.OffsetDateTime

import com.lessmarkup.interfaces.annotations.RequiredField
import com.lessmarkup.interfaces.data.{DataObject, OptionLong}

class Menu extends DataObject {
  @RequiredField
  var text: String = null
  @RequiredField
  var description: String = null
  var argument: String = null
  var uniqueId: String = null
  var order: Int = 0
  var visible: Boolean = false
  @RequiredField
  var created: OffsetDateTime = null
  var updated: OffsetDateTime = null
  var imageId: OptionLong = None
}
